### DISCUSSION TITLE
In the unit test of driver-can, how do we distinguish can from can-fd?

### DISCUSSION THREAD

**Discussion by SSQ1750** [zephyrproject-rtos/zephyr#76817] [Q&A] [answered]
On  tests/drivers/can/api/src/classic.c ， has a case: **Test that CAN-FD format frames are rejected in non-FD mode.**
And on prj.conf, CONFIG_CAN_FD_MODE=y has been set, means can-fd has enabled, then I am confused, so this case cannot be passed?Isn't this a conflict?

**Comment by henrikbrixandersen** [comment, marked as answer]
A CAN controller driver must reject CAN FD frames if its operational mode does not contain `CAN_MODE_FD`. That is what is being verified in that test case.

**Reply by SSQ1750** [reply]
Thanks,i got it ~
